OPHD/UI/Core/RadioButtonGroup.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+#include "TextControl.h"
+#include "Label.h"
+#include "../../Cache.h"
+#include "../../Constants.h"
+
+#include <NAS2D/Signal/Signal.h>
+#include <NAS2D/Signal/Delegate.h>
+#include <NAS2D/EventHandler.h>
+#include <NAS2D/Resource/Image.h>
+#include <NAS2D/Utility.h>
+#include <NAS2D/Renderer/Renderer.h>
+#include <NAS2D/MathUtils.h>
+
+#include <algorithm>
+#include <string>
+
+class RadioButtonGroup : public Control
+{
+public:
+	struct ButtonInfo
+	{
+		std::string name;
+		NAS2D::DelegateX<void> delegate;
+	};
+
+	RadioButtonGroup() = default;
+	RadioButtonGroup(std::vector<ButtonInfo> buttonInfos);
+	void setSelected(std::size_t index);
+	void onMove(NAS2D::Vector<int> displacement) override;
+	void update() override;
+	void clearSelection();
```

Comment:
Maybe group the `clearSelection` and `setSelected` methods together.

----

We should probably be consistent regarding "selection" versus "selected".

Reply:
I'm also of the opinion of not adding set/get to function names. E.g., _selection(std::size_t)_ ought to be fine.

That stated, I am aware that my brief naming convention like this has caused some ambiguity so this is open for debate. We should really come up with a standard for this and make it consistent across the code base (and probably hard define it in a developer doc).

Reply:
I was just thinking the `setSelected` method is an active method that changes state. Normally functions have verb names. What about naming it `select`?

Similarly, I would be ok with naming `clearSelection` as just `clear`. That's maybe a little more ambiguous, though in the context of a `RadioButtonGroup`, I imagine most people would be able to guess correctly.

----

Of course my above suggestion is perhaps getting away from the property naming syntax, which I think is what Leeor was originally going for. That is, name the property as a noun or adjective, and have both get and set forms of it, with a single name. I find that style more common in languages with property syntax support. In that case, reading or writing to a property is transparently redirected to function calls, without the code looking like a function call. I've seen it in VB, and I believe C# had similar support. I'm not so used to seeing it in C++. I wouldn't object to that style, just noting there is less syntax support for properties in C++.

Reply:
Renamed `setSelected` to `select` and renamed `clearSelection` to `clear`